repository/src/interpret/works.rs

```diff
@@ -0,0 +1,337 @@
+    for (remote, branch, commit_hash) in remote_tracking_branches {
+        result.push((format!("{remote}/{branch}"), sync(raw, commit_hash).await?));
+    }
+    Ok(result)
+}
+
+pub async fn clean(raw: &mut RawRepository, hard: bool) -> Result<(), Error> {
+    let finalized_branch_commit_hash = raw
+        .locate_branch(FINALIZED_BRANCH_NAME.into())
+        .await
+        .map_err(|e| match e {
+            raw::Error::NotFound(_) => {
+                eyre!(IntegrityError::new(
+                    "cannot locate `finalized` branch".to_string()
+                ))
+            }
+            _ => eyre!(e),
+        })?;
+    let branches = read_local_branches(raw).await?;
+    let last_header = read_last_finalized_block_header(raw).await?;
+    for (branch, branch_commit_hash) in branches {
+        if !(branch.as_str() == WORK_BRANCH_NAME
+            || branch.as_str() == FINALIZED_BRANCH_NAME
+            || branch.as_str() == FP_BRANCH_NAME)
+        {
+            if hard {
+                raw.delete_branch(branch.to_string()).await?;
+            } else {
+                // Delete outdated branch
+                let find_merge_base_result = raw
+                    .find_merge_base(branch_commit_hash, finalized_branch_commit_hash)
+                    .await
+                    .map_err(|e| match e {
+                        raw::Error::NotFound(_) => {
+                            eyre!(IntegrityError::new(format!(
+                                "cannot find merge base for branch {branch} and finalized branch"
+                            )))
+                        }
+                        _ => eyre!(e),
+                    })?;
+
+                if finalized_branch_commit_hash != find_merge_base_result {
+                    raw.delete_branch(branch.to_string()).await?;
+                }
+
+                // Delete branch with invalid commit sequence
+                raw.checkout(branch.to_string()).await?;
+                let reserved_state = raw.read_reserved_state().await?;
+                let commits =
+                    read_commits(raw, finalized_branch_commit_hash, branch_commit_hash).await?;
+                let mut verifier =
+                    CommitSequenceVerifier::new(last_header.clone(), reserved_state.clone())
+                        .map_err(|e| eyre!("failed to create a commit sequence verifier: {}", e))?;
+                for (commit, _) in commits.iter() {
+                    if verifier.apply_commit(commit).is_err() {
+                        raw.delete_branch(branch.to_string()).await?;
+                    }
+                }
+            }
+        }
+    }
+
+    // Remove remote repositories
+    // Note that remote branches are automatically removed when the remote repository is removed.
+    let remote_list = raw.list_remotes().await?;
+    for (remote_name, _) in remote_list {
+        raw.remove_remote(remote_name).await?;
+    }
+
+    Ok(())
+}
+
+pub async fn sync_old(
```

Comment:
Interface for `sync_old` is missing. Please add it with comments.

Reply:
It's legacy. Will be removed soon.